utilities.js

```diff
@@ -118,6 +118,76 @@ var Utilities = {
 		};
 	},
 
+	sentenceCase: function (text) {
+		const preserve = [];
+
+		// quoted text
+		text.replace(/“.*?”/g, (match, i) => {
+			preserve.push({ start: i, end: i + match.length });
+		});
+		text.replace(/‘.*?’/g, (match, i) => {
+			preserve.push({ start: i, end: i + match.length });
+		});
+		text.replace(/(["]).*?\1/g, (match, quote, i) => {
+			preserve.push({ start: i, end: i + match.length });
+		});
+
+		// sub-sentence start
+		text.replace(/([.?!][\s]+)(<[^>]+>)?([\p{Lu}])/ug, (match, end, markup, char, i) => {
+			markup = markup || "";
+			if (!text.substring(0, i + 1).match(/(\p{Lu}[.]){2,}$/u)) { // prevent "U.S. Taxes" from starting a new sub-sentence
+				preserve.push({ start: i + end.length + markup.length, end: i + end.length + markup.length + char.length });
+			}
+		});
+
+		// protect leading capital
+		text.replace(/^(<[^>]+>)?([\p{Lu}])/u, (match, markup, char) => {
+			markup = markup || "";
+			preserve.push({ start: markup.length, end: markup.length + char.length });
+		});
+
+		// protect nocase
+		text.replace(/<span class="nocase">.*?<\/span>|<nc>.*?<\/nc>/gi, (match, i) => {
+			preserve.push({ start: i, end: i + match.length, description: 'nocase' });
+		});
+
+		// mask html tags with characters so the sentence-casing can deal with them as simple words
+		let masked = text.replace(/<[^>]+>/g, (match, i) => {
+			preserve.push({ start: i, end: i + match.length, description: 'markup' });
+			return '\uFFFD'.repeat(match.length);
+		});
```

Comment:
The above two sections don't seem to be doing anything, at least with the sample input. Do we need them? If so, we should add some test data for them.

Reply:
The (sub-)sentence-start ones? One of them did fail a test when I removed them, but I've added two new samples. But in the mail I received you seemed to be pointing towards the (sub-)sentence start handlers, here on the site it looks like you're referring to the markup handler. I'll look into the markup handler tonight -- certainly *that* should hit on something.

Reply:
The two blocks above my comment here — `protect nocase` and `mask html tags with characters…`. Nothing failed when I removed those (but still setting `masked` properly).

Reply:
Well that's just weird. I'll look into it tonight - I have a 6 hour drive in front of me right now.

Reply:
It captures in-word markup. I will grant that I do not have a non-synthetic case at hand, but I've added one.